Sources/TuistAutomation/Device/PhysicalDevice.swift

```diff
@@ -6,7 +6,13 @@ public struct PhysicalDevice: Codable, Equatable, Identifiable {
     public let id: String
     public let name: String
     public let platform: Platform
-    public let osVersion: String
+    public let osVersion: String?
+}
+
+extension PhysicalDevice {
+    public var isReachable: Bool {
+        osVersion != nil
+    }
```

Comment:
I'd add a property called `state` in the `PhysicalDevice` instead of assuming whether a device is reachable based on `osVersion` being nil or not:
```swift
public struct PhysicalDevice {
  public enum ConnectionState {
    case wifi, usb, unreachable
  }

  public let connectionState: ConnectionState
}
```

We can derive the proper `connectionState` when initializing the `PhysicalDevice` from the `devicectl` output by using the `transportType` property and `ddiServicesAvailable` (if that one is needed)

Reply:
If we use the connection state, how about designing it in the style below? The `tunnelState` property also needed.

```swift
public struct PhysicalDevice {
    public enum TransportType {
        case wifi
        case usb
        case unknown
    }

    public enum ConnectionState {
        case connected
        case disconnected
    }

    public let transportType: TransportType
    public let connectionState: ConnectionState
}
```

<img src="https://github.com/user-attachments/assets/0d6576ad-beb6-4793-b77d-e5651559be73" alt="Description" width="400"/>

Reply:
This looks amazing, thousand times yes!

Reply:
Related changes have been pushed; please review again.